ui.admin-mui/src/components/Settings/Import/nightbot.tsx

```diff
@@ -65,29 +66,156 @@ const PageSettingsModulesImportNightbot: React.FC<{
+    const ax = axios.create({
+      signal:  controller.signal,
+      headers: { Authorization: 'Bearer ' + accessToken },
+    });
+
+    // NOTE: This attempts to jump in when the request is bad
+    //       Do NOT handle 429 like this? Just self-impose rate limits?
+    ax.interceptors.request.use((config) => {
+      return config;
+    }, (error) => {
+      enqueueSnackbar('Something went wrong.', { variant: 'error' });
+      console.log('[Bad request] - ', error);
+      // NOTE: Isn't it too late to abort here because by now we already
+      //       have an error response to handle instead?
+      //       The desire is to cascade through any pending requests
+      //       and abort them all, but I don't think this works
+      controller.abort();
+      return Promise.reject(error);
+    });
+
+    // NOTE: key is the name of the path entry to the resource we want
+    //       For just the playlist, this is fine. If we add more resources
+    //       this key will need to be dynamic
+    const key = 'playlist';
+
+    const fetchResource = async (acc: PlaylistItemTrack[], offset: number): Promise<T> => {
+      // NOTE: This breaks on any other endpoint now
+      // NOTE: We probably could dispatch to a handler on the `key` aka the resource type
+      const response = await ax.get('https://api.nightbot.tv/1/song_requests/playlist?limit=100&offset='+offset, {
+        signal:  controller.signal,
+        headers: { Authorization: 'Bearer ' + accessToken },
+      });
+
+      // NOTE: This appeases the type checker when mapping over `data[key]`
+      const data: ApiResponse = response.data;
+
+      // NOTE: I think a default of 0 is safe, to ensure requesting exactly 1 page of data
+      const total = data._total ?? 0;
+
+      // NOTE: Resources with multiple items are returned in an array
+      //       Fix this if we expand the importer to other endpoints
+      const tracks = data[key].map((e) => e.track);
+
+      // TODO: I'd prefer not to mutate this accumulator in-place
+      if (acc.push(...tracks) >= total) {
+        return acc;
+      }
+
+      return await fetchResource(acc, offset + 100);
+    };
+
+    const pl = await fetchResource([], 0);
+    setPlaylist(pl);
+
+    return pl;
+  };
+
+  const importPlaylist = async () => {
+    // TODO: `fetchPlaylist` should probably just return the simplified array of IDs
+    const videos: PlaylistItemTrack[] = await fetchPlaylist();
+
+    videos.filter((track) => {
+      // TODO: Nightbot supports SoundCloud also
+      //       Should we cover that case?
+      track.provider === 'youtube'
+    }).forEach(async (video) => {
+      await new Promise((resolve, reject) => {
+        getSocket('/systems/songs').emit(
+          'import.video',
+          {
+            playlist:  video.providerId,
+            forcedTag: 'nightbot-import',
+          },
+          (err) => {
+            if (err) {
+              // FIXME: This could potentially generate
```

Comment:
you can just count number of failures and if >0 then just enqueueSnackbar something like `x video(s) failed to import` or something like that. wdyt?